packages/atclient/include/atclient/atclient.h

```diff
@@ -123,8 +123,7 @@ void atclient_stop_atserver_connection(atclient *ctx);
  * @param atsign the atsign the atkeys belong to, this string is assumed to be null terminated
  * @return int 0 on success, non-zero on error
  */
-int atclient_pkam_authenticate(atclient *ctx, const char *atserver_host, const int atserver_port,
-                               const atclient_atkeys *atkeys, const char *atsign);
+int atclient_pkam_authenticate(atclient *ctx, const atclient_atkeys *atkeys, const char *atsign, atclient_pkam_authenticate_options *options);
```

Comment:
I would make the function signature this:

```
int atclient_pkam_authenticate(atclient *ctx, const char *atsign, const atclient_atkeys *atkeys,, atclient_pkam_authenticate_options *options);
```

Because `atkeys` may become an optional argument in the future, so let's put it near the end.